src/main/java/org/mifos/pheevouchermanagementsystem/api/implementation/VoucherLifecycleManagementApiController.java

```diff
@@ -17,16 +22,53 @@
 public class VoucherLifecycleManagementApiController implements VoucherLifecycleManagementApi {
     @Autowired
     ActivateVoucherService activateVoucherService;
+    @Autowired
+    RedeemVoucherService redeemVoucherService;
+    @Autowired
+    ObjectMapper objectMapper;
+    @Autowired
+    VoucherLifecycleService voucherLifecycleService;
     @Override
-    public ResponseDTO voucherStatusChange(String callbackURL, RequestDTO requestBody, String command) throws ExecutionException, InterruptedException, JsonProcessingException {
+    public <T> ResponseEntity<T> voucherStatusChange(String callbackURL, String registeringInstitutionId, Object requestBody, String command) throws ExecutionException, InterruptedException, JsonProcessingException {
+        RequestDTO requestDTO = null;
+        RedeemVoucherRequestDTO redeemVoucherRequestDTO = null;
         try {
             if(command.equals("activate")){
-                activateVoucherService.activateVouchers(requestBody, callbackURL);
-            }        } catch (Exception e) {
-            return new ResponseDTO(FAILED_RESPONSE.getValue(), FAILED_RESPONSE.getMessage(), requestBody.getRequestID());
-
+                requestDTO = objectMapper.convertValue(requestBody, RequestDTO.class);
+                activateVoucherService.activateVouchers(requestDTO, callbackURL, registeringInstitutionId);
+            }
+            else if(command.equals("redeem")){
+                redeemVoucherRequestDTO = objectMapper.convertValue(requestBody, RedeemVoucherRequestDTO.class);
+                return ResponseEntity.status(HttpStatus.OK).body((T) redeemVoucherService.redeemVoucher(redeemVoucherRequestDTO, registeringInstitutionId));
+            }
+            else if(command.equals("suspend")){
+                requestDTO = objectMapper.convertValue(requestBody, RequestDTO.class);
+                voucherLifecycleService.suspendVoucher(requestDTO, callbackURL, registeringInstitutionId);
+            }
+            else if(command.equals("reactivate")){
+                requestDTO = objectMapper.convertValue(requestBody, RequestDTO.class);
+                voucherLifecycleService.reactivateVoucher(requestDTO, callbackURL, registeringInstitutionId);
+            }
+            else if(command.equals("cancel")){
+                requestDTO = objectMapper.convertValue(requestBody, RequestDTO.class);
+                voucherLifecycleService.cancelVoucher(requestDTO, callbackURL, registeringInstitutionId);
+            }
+            else if(command.equals("redeemPay")){
```

Comment:
add an else case and throw unsupported operation exception

Reply:
redeemPay will be async because it invokes a zeebe workflow